pkg/inventory/model/client.go

```diff
@@ -667,27 +667,29 @@ type Labeler struct {
 // Insert labels for the model into the DB.
 func (r *Labeler) Insert(table Table, model Model) error {
 	kind := table.Name(model)
-	for l, v := range model.Labels() {
-		label := &Label{
-			Parent: model.Pk(),
-			Kind:   kind,
-			Name:   l,
-			Value:  v,
-		}
-		err := table.Insert(label)
-		if err != nil {
-			return err
+	if labeled, cast := model.(Labeled); cast {
```

Comment:
Labels() no longer part of the `Model` interface.  Instead the model implements the `Labeled` interface as needed.